### PR TITLE
feat: upsells checkout a/b test

### PR DESCRIPTION
**Pull request by egoroza**
implements upsells a/b test on checkout page

### CODE REVIEW

File: src/pages/Checkout/CheckoutPage.vue [hunk 1]
```diff
@@ -323,6 +324,7 @@ export default {
 			verificationSubmitted: false,
 			showVerification: false,
 			showVerifyRemovePromoCredit: false,
+			isUpsellsExperimentEnabled: false,
```
Comment (mcstover) [inline src/pages/Checkout/CheckoutPage.vue:327]: What will this flag control?

File: src/pages/Checkout/CheckoutPage.vue [hunk 2]
```diff
@@ -354,11 +356,24 @@ export default {
...
 				.then(() => {
 					return Promise.all([
-						client.query({ query: initializeCheckout, fetchPolicy: 'network-only' })
+						client.query({ query: initializeCheckout, fetchPolicy: 'network-only' }),
+						client.query({ query: experimentQuery, variables: { id: 'upsells_checkout' } }),
 					]);
 				});
 		},
 		result({ data }) {
+			const upsellsExperiment = this.apollo.readFragment({
```
Comment (mcstover) [inline src/pages/Checkout/CheckoutPage.vue:365]: Please move all of this new code from the `result` block to either `created` or `mounted`

File: src/pages/Checkout/CheckoutPage.vue (hunk 1 above)
Comment (mcstover) [reply src/pages/Checkout/CheckoutPage.vue:327]: I see the design now. Looks nice!

**Review by mcstover — APPROVED** [review]

**Comment by egoroza** [comment]
noticed that the `Experiment:upsells_checkout` query returns a response, but returns `null` for the version 🤔  unsure why...

File: src/graphql/query/checkout/initializeCheckout.graphql [hunk 3]
```diff
@@ -30,6 +30,10 @@ query initializeCheckout($basketId: String) {
 			value
 			description
 		}
+		upsellsEnabled: uiExperimentSetting(key: "upsells_checkout") {
```
Comment (mcstover) [inline src/graphql/query/checkout/initializeCheckout.graphql:33]: @egoroza Please move this Experiment setting to https://github.com/kiva/ui/blob/main/src/graphql/query/checkout/checkoutSettings.graphql and things will start to work.

The `initializeCheckout` is running a network only call and bypassing the cache. Experiment assignment depends on the setting existing in the cache.

Comment (mcstover) [reply src/graphql/query/checkout/initializeCheckout.graphql:33]: @egoroza Please remove this additional query block from this file.

**Pull request merged**